### PR TITLE
Add msgpack support to jerakia api

### PR DESCRIPTION
**Pull request by hhenkel**
This PR is related to [issue jerakia/#95](https://github.com/crayfishx/jerakia/issues/95) and [PR jerakia/#96](https://github.com/crayfishx/jerakia/issues/96)

### CODE REVIEW

File: lib/jerakia/client.rb [hunk 1]
```diff
@@ -93,22 +99,31 @@ def http_send(request)
...
       return http_send(request)
     end
 
     def put(url_path, params)
       uri = URI.parse(url_address + url_path)
       request = Net::HTTP::Put.new(uri.path)
-      request.add_field('Content-Type', 'application/json')
-      request.body = params.to_json
+      request = set_content_type(request)
+      request.body = params.method(@method_name).call
```
Comment (crayfishx) [inline lib/jerakia/client.rb]: See the comment on your PR in Jerakia core, couldnt we do it the same way here?  `request.body = encode_params(params)` or something?

File: lib/jerakia/client.rb [hunk 2]
```diff
@@ -93,22 +99,39 @@ def http_send(request)
...
+        if request.key?('body')
+          request.body.to_json
+        end
+      elsif @config[:content_type] == 'msgpack'
+        request.add_field('Content-Type', 'application/x-msgpack')
+        if request.key?('body')
+          request.body.to_msgpack
+        end
+      else
+        STDERR.puts("Invalid setting \"#{@config[:content_type]}\" for \":content_type\" in config - supported is either json or msgpack.")
```
Comment (crayfishx) [inline lib/jerakia/client.rb]: Is there a reason you need to write direct to STDERR rather than `raise Jerakia::Client::Error(.....)`?

Comment (crayfishx) [reply lib/jerakia/client.rb]: My experience with Python yesterday has clearly affected me - I meant of course `raise Jerakia::Client::Error, "....."`

Comment (hhenkel) [reply lib/jerakia/client.rb]: My lacking knowledge of ruby and my laziness of digging deeper was the cause... ;) I'll update the code...

**Pull request merged**